learning_journal/learning_journal/views/default.py

```diff
@@ -0,0 +1,73 @@
+from pyramid.response import Response
+from pyramid.view import view_config
+from pyramid.httpexceptions import HTTPNotFound
+from pyramid.httpexceptions import HTTPFound
+import datetime
+
+from sqlalchemy.exc import DBAPIError
+
+from ..models import MyEntry
+
+
+@view_config(route_name="home", renderer="../templates/home.jinja2")
+def home_view(request):
+    try:
+        query = request.dbsession.query(MyEntry)
+        entries = query.all()
+    except DBAPIError:
+        return Response(db_err_msg, content_type="text/plain", status=500)
+    return {"entries": entries}
+
+
+@view_config(route_name='detail', renderer='../templates/detail.jinja2')
+def detail_view(request):
+    entry_id = int(request.matchdict['id'])
+    entry = request.dbsession.query(MyEntry).get(entry_id)
+    return {"entry": entry}
```

Comment:
If entry is not found by id here, this is going to raise a template rendering error, because `entry` will be the `None` type object.  You don't want that.  YOu should be returning a `404 Not Found` response if you can't find the entry they asked for by id.